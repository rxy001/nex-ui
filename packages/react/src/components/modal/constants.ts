export const MODAL_INTERNAL_ID_PREFIX = 'nui-modal-'
