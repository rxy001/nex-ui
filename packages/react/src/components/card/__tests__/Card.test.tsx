import {
  testComponentStability,
  renderWithNexUIProvider,
  testVariantDataAttrs,
  testRefForwarding,
  testRootClassName,
} from '~/tests/shared'
import { cardClasses, cardDataAttrs } from './constants'
import { Card } from '../index'

describe('Card', () => {
  testComponentStability(<Card />)

  testVariantDataAttrs(<Card />, ['shadow', ['xs', 'sm', 'md', 'lg', 'xl']])

  testVariantDataAttrs(<Card />, ['radius', ['none', 'sm', 'md', 'lg']])

  testVariantDataAttrs(<Card />, ['blurred', [true, false]])

  testVariantDataAttrs(<Card />, ['hoverable', [true, false]])

  testRefForwarding(<Card />)

  testRootClassName(<Card />)

  it('should render with default props', () => {
    const { container } = renderWithNexUIProvider(<Card />)
    const card = container.firstElementChild

    expect(card).toHaveClass(cardClasses.root)
    expect(card).toHaveAttribute(...cardDataAttrs['shadow-md'])
    expect(card).toHaveAttribute(...cardDataAttrs['radius-md'])
    expect(card).toHaveAttribute(...cardDataAttrs['hoverable-false'])
    expect(card).toHaveAttribute(...cardDataAttrs['blurred-false'])

    expect(card).toMatchSnapshot()
  })
})
