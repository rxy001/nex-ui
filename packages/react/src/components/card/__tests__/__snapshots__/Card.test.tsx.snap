// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render with default props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: var(--nui-colors-content);
  max-width: 345px;
  position: relative;
  overflow: hidden;
  box-shadow: var(--nui-shadows-md);
  border-radius: var(--nui-radii-lg);
}

<div
  class="nui-card-root emotion-0"
  data-blurred="false"
  data-hoverable="false"
  data-radius="md"
  data-shadow="md"
/>
`;
